src/stavenote.ts

```diff
@@ -589,44 +653,46 @@ export class StaveNote extends StemmableNote {
   }
 
   // Determine if current note is a rest
-  isRest() {
+  isRest(): boolean {
     return this.glyph.rest;
   }
 
   // Determine if the current note is a chord
-  isChord() {
+  isChord(): boolean {
     return !this.isRest() && this.keys.length > 1;
   }
 
   // Determine if the `StaveNote` has a stem
-  hasStem() {
+  hasStem(): boolean {
     return this.glyph.stem;
   }
 
-  hasFlag() {
+  hasFlag(): boolean {
     return super.hasFlag() && !this.isRest();
   }
 
-  getStemX() {
+  getStemX(): number {
     if (this.noteType === 'r') {
       return this.getCenterGlyphX();
     } else {
       // We adjust the origin of the stem because we want the stem left-aligned
       // with the notehead if stemmed-down, and right-aligned if stemmed-up
-      return super.getStemX() + getStemAdjustment(this);
+      return super.getStemX() + (this.stem_direction ? Stem.WIDTH / (2 * -this.stem_direction) : 0);
```

Comment:
Why did you replace `getStemAdjustment`?

Reply:
This was a global non exported const function in this file used only once here. Within this function that did not belonged to the class it was also not possible to check if `this.stem_direction` was not `undefined`: 

```
const getStemAdjustment = (note) => Stem.WIDTH / (2 * -note.getStemDirection());
```

Reply:
Okay, thanks, makes sense.